src/util/worker/future.rs

```diff
@@ -55,7 +55,7 @@ pub struct Scheduler<T> {
 }
 
 impl<T: Display> Scheduler<T> {
-    fn new<S: Into<String>>(name: S, sender: UnboundedSender<Option<T>>) -> Scheduler<T> {
+    pub fn new<S: Into<String>>(name: S, sender: UnboundedSender<Option<T>>) -> Scheduler<T> {
```

Comment:
You can revert this change now since we are not calling `Scheduler::new` any more. Rest LGTM.